File: C++/Add Two Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        
        ListNode ans;
        ListNode* calc = &ans;

        int s_carry = 0;

        while (l1 || l2 || s_carry) {
            
            s_carry = (l1 ? l1->val : 0) + (l2 ? l2->val : 0) + s_carry;

            calc->next = new ListNode(s_carry % 10);
            calc = calc->next;

            s_carry = s_carry / 10;

            if (l1) l1 = l1->next;
            if (l2) l2 = l2->next;

        }

        return ans.next;

    }
};

File: python/Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        
        t = l1

        x = 0
        y = 0
        i = 0

        #* Convert to integer
        while t != None:
            
            x += t.val * (10**i)

            t = t.next
            i += 1

        t = l2
        i = 0

        #* Convert to integer
        while t != None:
            
            y += t.val * (10**i)

            t = t.next
            i += 1

        ans = x + y
        __res = ListNode()
        res = __res

        #* Reverse the answer and store in a linked list.
        while (ans > 0):

            tmp = ListNode()
            __res.val = ans % 10
            ans = ans // 10

            if ans > 0:
                __res.next = tmp
                __res = __res.next

        return res
